src/python/module/smartredis/configoptions.py

```diff
@@ -0,0 +1,225 @@
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .smartredisPy import PyConfigOptions
+from .util import exception_handler, typecheck
+
+from .error import RedisRuntimeError
+
+_notfactory = "Method called on a ConfigOptions object not created from a factory method"
+
+class ConfigOptions:
+    def __init__(self):
+        """Initialize a ConfigOptions base object
+        """
+        self._is_created_via_factory = False
+        self._config_opts = None
+
+    @staticmethod
+    def from_pybind(configoptions):
+        """Initialize a ConfigOptions object from
+        a PyConfigOptions object
+
+        :param configoptions: The pybind PyConfigOptions object
+                              to use for construction
+        :type dataset: PyConfigOptions
+        :return: The newly constructed ConfigOptions from
+                 the PyConfigOptions
+        :rtype: ConfigOptions
+        """
+        typecheck(configoptions, "configoptions", PyConfigOptions)
+        new_configoptions = ConfigOptions()
+        new_configoptions.set_configoptions(configoptions)
+        return new_configoptions
+
+    @exception_handler
+    def set_configoptions(self, configoptions):
+        """Set the PyConfigOptions attribute
+
+        :param configoptions: The PyConfigOptions object
+        :type configoptions: PyConfigOptions
+        """
+        typecheck(configoptions, "configoptions", PyConfigOptions)
+        self._config_opts = configoptions
+
+    def _is_factory_object(self):
+        """Check whether this object was created via a factory method
+        """
+        return self._is_created_via_factory
+
+    @classmethod
+    @exception_handler
+    def create_from_environment(cls, db_prefix):
+        """Instantiate ConfigOptions, getting selections from
+        environment variables. If db_prefix is non-empty,
+        then "{db_prefix}_" will be prepended to the name of
+        each environment variable that is read
+
+        :param cls: The ConfigOptions class
+        :type cls: type
+        :param db_prefix: Prefix to prepend to environment variables
+                          or an empty string to eschew prepending
+        :type db_prefix: str
+        :return: An instantiated ConfigOptions object
+        :rtype: ConfigOptions
+        """
+        typecheck(db_prefix, "db_prefix", str)
+        factory_object = PyConfigOptions.create_from_environment(db_prefix)
+        result = cls.from_pybind(factory_object)
+        result._is_created_via_factory = True
+        return result
+
+    # Configuration via JSON file or JSON blob is anticipated in the future
+    # but not supported yet
+    if False:
```

Comment:
Sounds like the majority opinion is no stubs: just a note here to remember to remove

Reply:
done